boxoffice/models/discount_policy.py

```diff
@@ -53,6 +58,33 @@ def is_automatic(self):
     def is_coupon(self):
         return self.discount_type == DISCOUNT_TYPE.COUPON
 
+    def gen_signed_code(self):
+        """Generates a signed code in the format discount_code_base.randint.signature"""
+        signer = Signer(self.secret)
+        key = "{base}.{randint}".format(base=self.discount_code_base, randint=random.randint(1, 10000))
```

Comment:
Don't hardcode this for randint. Put that in the view. We'll have two scenarios:

1. A company wants a few thousand codes, so we make them sequence numbers 1..1000. The policy is linked to their account (once Lastuser integration is in place), so they can login to the dashboard and copy codes over or re-download them.

2. We're mailing our database with a referral code. As the sequence numbers here give away the size of the database, or your position number, we use randint. However, because it's randint, there's no way to track who was given what number, or to retrieve that number. These codes can be generated once only. There's no other way to ensure two different people weren't given the same code. Finally, because you can't use the same randint twice even within one batched call, the caller needs to generate the random numbers and populate a dictionary with {randint: generated code}.